Cloned workflow is not saved

When a workflow is cloned and then saved, database object is not written, and an error message will be logged:

```
[DEBUG] 2021-06-29 12:43:07.219 [http-nio-8080-exec-8] SearchService - org.kitodo.data.database.exceptions.DAOException: javax.persistence.PersistenceException: org.hibernate.HibernateException: Don't change the reference to a collection with delete-orphan enabled : org.kitodo.data.database.beans.Workflow.templates
[ERROR] 2021-06-29 12:43:07.219 [http-nio-8080-exec-8] WorkflowForm - org.kitodo.data.database.exceptions.DAOException: javax.persistence.PersistenceException: org.hibernate.HibernateException: Don't change the reference to a collection with delete-orphan enabled : org.kitodo.data.database.beans.Workflow.templates
org.kitodo.data.exceptions.DataException: org.kitodo.data.database.exceptions.DAOException: javax.persistence.PersistenceException: org.hibernate.HibernateException: Don't change the reference to a collection with delete-orphan enabled : org.kitodo.data.database.beans.Workflow.templates
	at org.kitodo.production.services.data.base.SearchService.save(SearchService.java:287) ~[classes/:3.3]
	at org.kitodo.production.forms.WorkflowForm.saveWorkflow(WorkflowForm.java:290) [classes/:3.3]
	at org.kitodo.production.forms.WorkflowForm.saveAndRedirect(WorkflowForm.java:146) [classes/:3.3]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_282]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_282]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_282]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_282]
	at org.apache.el.parser.AstValue.invoke(AstValue.java:247) [jasper-el.jar:9.0.43]
	at org.apache.el.MethodExpressionImpl.invoke(MethodExpressionImpl.java:267) [jasper-el.jar:9.0.43]
	at org.jboss.weld.util.el.ForwardingMethodExpression.invoke(ForwardingMethodExpression.java:40) [weld-servlet-2.4.8.Final.jar:2.4.8.Final]
	at org.jboss.weld.el.WeldMethodExpression.invoke(WeldMethodExpression.java:50) [weld-servlet-2.4.8.Final.jar:2.4.8.Final]
	at org.jboss.weld.util.el.ForwardingMethodExpression.invoke(ForwardingMethodExpression.java:40) [weld-servlet-2.4.8.Final.jar:2.4.8.Final]
	at org.jboss.weld.el.WeldMethodExpression.invoke(WeldMethodExpression.java:50) [weld-servlet-2.4.8.Final.jar:2.4.8.Final]
	at org.apache.myfaces.view.facelets.el.ContextAwareTagMethodExpression.invoke(ContextAwareTagMethodExpression.java:96) [myfaces-impl-2.3.4.jar:2.3.4]
	at org.apache.myfaces.application.ActionListenerImpl.processAction(ActionListenerImpl.java:74) [myfaces-impl-2.3.4.jar:2.3.4]
	at org.springframework.faces.webflow.FlowActionListener.processAction(FlowActionListener.java:71) [spring-faces-2.5.1.RELEASE.jar:2.5.1.RELEASE]
	at org.springframework.faces.model.SelectionTrackingActionListener.processAction(SelectionTrackingActionListener.java:64) [spring-faces-2.5.1.RELEASE.jar:2.5.1.RELEASE]
	at javax.faces.component.UICommand.broadcast(UICommand.java:120) [myfaces-api-2.3.4.jar:2.3.4]
	at javax.faces.component.UIViewRoot._broadcastAll(UIViewRoot.java:1255) [myfaces-api-2.3.4.jar:2.3.4]
	at javax.faces.component.UIViewRoot.broadcastEvents(UIViewRoot.java:420) [myfaces-api-2.3.4.jar:2.3.4]
	at javax.faces.component.UIViewRoot._process(UIViewRoot.java:1741) [myfaces-api-2.3.4.jar:2.3.4]
	at javax.faces.component.UIViewRoot.processApplication(UIViewRoot.java:935) [myfaces-api-2.3.4.jar:2.3.4]
	at org.apache.myfaces.lifecycle.InvokeApplicationExecutor.execute(InvokeApplicationExecutor.java:42) [myfaces-impl-2.3.4.jar:2.3.4]
	at org.apache.myfaces.lifecycle.LifecycleImpl.executePhase(LifecycleImpl.java:195) [myfaces-impl-2.3.4.jar:2.3.4]
	at org.apache.myfaces.lifecycle.LifecycleImpl.execute(LifecycleImpl.java:142) [myfaces-impl-2.3.4.jar:2.3.4]
	at javax.faces.webapp.FacesServlet.service(FacesServlet.java:204) [myfaces-api-2.3.4.jar:2.3.4]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:227) [catalina.jar:9.0.43]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162) [catalina.jar:9.0.43]
	at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:53) [tomcat-websocket.jar:9.0.43]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189) [catalina.jar:9.0.43]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162) [catalina.jar:9.0.43]
	at org.kitodo.production.servletfilter.EncodingFilter.doFilter(EncodingFilter.java:69) [classes/:3.3]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189) [catalina.jar:9.0.43]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162) [catalina.jar:9.0.43]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:317) [spring-security-web-4.2.13.RELEASE.jar:4.2.13.RELEASE]
	at org.kitodo.production.security.SecurityObjectAccessFilter.doFilter(SecurityObjectAccessFilter.java:56) [classes/:3.3]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331) [spring-security-web-4.2.13.RELEASE.jar:4.2.13.RELEASE]
	at org.springframework.security.web.access.intercept.FilterSecurityInterceptor.invoke(FilterSecurityInterceptor.java:127) [spring-security-web-4.2.13.RELEASE.jar:4.2.13.RELEASE]
	at org.springframework.security.web.access.intercept.FilterSecurityInterceptor.doFilter(FilterSecurityInterceptor.java:91) [spring-security-web-4.2.13.RELEASE.jar:4.2.13.RELEASE]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331) [spring-security-web-4.2.13.RELEASE.jar:4.2.13.RELEASE]
	at org.springframework.security.web.access.ExceptionTranslationFilter.doFilter(ExceptionTranslationFilter.java:114) [spring-security-web-4.2.13.RELEASE.jar:4.2.13.RELEASE]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331) [spring-security-web-4.2.13.RELEASE.jar:4.2.13.RELEASE]
	at org.springframework.security.web.session.SessionManagementFilter.doFilter(SessionManagementFilter.java:137) [spring-security-web-4.2.13.RELEASE.jar:4.2.13.RELEASE]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331) [spring-security-web-4.2.13.RELEASE.jar:4.2.13.RELEASE]
	at org.springframework.security.web.authentication.AnonymousAuthenticationFilter.doFilter(AnonymousAuthenticationFilter.java:111) [spring-security-web-4.2.13.RELEASE.jar:4.2.13.RELEASE]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331) [spring-security-web-4.2.13.RELEASE.jar:4.2.13.RELEASE]
	at org.springframework.security.web.servletapi.SecurityContextHolderAwareRequestFilter.doFilter(SecurityContextHolderAwareRequestFilter.java:170) [spring-security-web-4.2.13.RELEASE.jar:4.2.13.RELEASE]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331) [spring-security-web-4.2.13.RELEASE.jar:4.2.13.RELEASE]
	at org.springframework.security.web.savedrequest.RequestCacheAwareFilter.doFilter(RequestCacheAwareFilter.java:63) [spring-security-web-4.2.13.RELEASE.jar:4.2.13.RELEASE]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331) [spring-security-web-4.2.13.RELEASE.jar:4.2.13.RELEASE]
	at org.springframework.security.web.session.ConcurrentSessionFilter.doFilter(ConcurrentSessionFilter.java:155) [spring-security-web-4.2.13.RELEASE.jar:4.2.13.RELEASE]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331) [spring-security-web-4.2.13.RELEASE.jar:4.2.13.RELEASE]
	at org.springframework.security.web.authentication.AbstractAuthenticationProcessingFilter.doFilter(AbstractAuthenticationProcessingFilter.java:200) [spring-security-web-4.2.13.RELEASE.jar:4.2.13.RELEASE]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331) [spring-security-web-4.2.13.RELEASE.jar:4.2.13.RELEASE]
	at org.springframework.security.web.authentication.logout.LogoutFilter.doFilter(LogoutFilter.java:116) [spring-security-web-4.2.13.RELEASE.jar:4.2.13.RELEASE]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331) [spring-security-web-4.2.13.RELEASE.jar:4.2.13.RELEASE]
	at org.springframework.security.web.header.HeaderWriterFilter.doFilterInternal(HeaderWriterFilter.java:66) [spring-security-web-4.2.13.RELEASE.jar:4.2.13.RELEASE]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107) [spring-web-4.3.23.RELEASE.jar:4.3.23.RELEASE]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331) [spring-security-web-4.2.13.RELEASE.jar:4.2.13.RELEASE]
	at org.springframework.security.web.context.SecurityContextPersistenceFilter.doFilter(SecurityContextPersistenceFilter.java:105) [spring-security-web-4.2.13.RELEASE.jar:4.2.13.RELEASE]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331) [spring-security-web-4.2.13.RELEASE.jar:4.2.13.RELEASE]
	at org.springframework.security.web.context.request.async.WebAsyncManagerIntegrationFilter.doFilterInternal(WebAsyncManagerIntegrationFilter.java:56) [spring-security-web-4.2.13.RELEASE.jar:4.2.13.RELEASE]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107) [spring-web-4.3.23.RELEASE.jar:4.3.23.RELEASE]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331) [spring-security-web-4.2.13.RELEASE.jar:4.2.13.RELEASE]
	at org.springframework.security.web.FilterChainProxy.doFilterInternal(FilterChainProxy.java:214) [spring-security-web-4.2.13.RELEASE.jar:4.2.13.RELEASE]
	at org.springframework.security.web.FilterChainProxy.doFilter(FilterChainProxy.java:177) [spring-security-web-4.2.13.RELEASE.jar:4.2.13.RELEASE]
	at org.springframework.web.filter.DelegatingFilterProxy.invokeDelegate(DelegatingFilterProxy.java:347) [spring-web-4.3.23.RELEASE.jar:4.3.23.RELEASE]
	at org.springframework.web.filter.DelegatingFilterProxy.doFilter(DelegatingFilterProxy.java:263) [spring-web-4.3.23.RELEASE.jar:4.3.23.RELEASE]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189) [catalina.jar:9.0.43]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162) [catalina.jar:9.0.43]
	at org.apache.logging.log4j.web.Log4jServletFilter.doFilter(Log4jServletFilter.java:71) [log4j-web-2.13.2.jar:2.13.2]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189) [catalina.jar:9.0.43]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162) [catalina.jar:9.0.43]
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:202) [catalina.jar:9.0.43]
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:97) [catalina.jar:9.0.43]
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:542) [catalina.jar:9.0.43]
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:143) [catalina.jar:9.0.43]
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:92) [catalina.jar:9.0.43]
	at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:687) [catalina.jar:9.0.43]
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:78) [catalina.jar:9.0.43]
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:346) [catalina.jar:9.0.43]
	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:374) [tomcat-coyote.jar:9.0.43]
	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:65) [tomcat-coyote.jar:9.0.43]
	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:887) [tomcat-coyote.jar:9.0.43]
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1684) [tomcat-coyote.jar:9.0.43]
	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49) [tomcat-coyote.jar:9.0.43]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_282]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_282]
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61) [tomcat-util.jar:9.0.43]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_282]
Caused by: org.kitodo.data.database.exceptions.DAOException: javax.persistence.PersistenceException: org.hibernate.HibernateException: Don't change the reference to a collection with delete-orphan enabled : org.kitodo.data.database.beans.Workflow.templates
	at org.kitodo.data.database.persistence.BaseDAO.storeObject(BaseDAO.java:354) ~[kitodo-data-management-3.3.jar:?]
	at org.kitodo.data.database.persistence.BaseDAO.save(BaseDAO.java:88) ~[kitodo-data-management-3.3.jar:?]
	at org.kitodo.production.services.data.base.SearchDatabaseService.saveToDatabase(SearchDatabaseService.java:64) ~[classes/:3.3]
	at org.kitodo.production.services.data.base.SearchService.save(SearchService.java:278) ~[classes/:3.3]
	... 89 more
Caused by: javax.persistence.PersistenceException: org.hibernate.HibernateException: Don't change the reference to a collection with delete-orphan enabled : org.kitodo.data.database.beans.Workflow.templates
	at org.hibernate.internal.ExceptionConverterImpl.convert(ExceptionConverterImpl.java:154) ~[hibernate-core-5.4.18.Final.jar:5.4.18.Final]
	at org.hibernate.internal.ExceptionConverterImpl.convert(ExceptionConverterImpl.java:181) ~[hibernate-core-5.4.18.Final.jar:5.4.18.Final]
	at org.hibernate.internal.ExceptionConverterImpl.convert(ExceptionConverterImpl.java:188) ~[hibernate-core-5.4.18.Final.jar:5.4.18.Final]
	at org.hibernate.internal.SessionImpl.doFlush(SessionImpl.java:1364) ~[hibernate-core-5.4.18.Final.jar:5.4.18.Final]
	at org.hibernate.internal.SessionImpl.flush(SessionImpl.java:1347) ~[hibernate-core-5.4.18.Final.jar:5.4.18.Final]
	at org.kitodo.data.database.persistence.BaseDAO.storeObject(BaseDAO.java:351) ~[kitodo-data-management-3.3.jar:?]
	at org.kitodo.data.database.persistence.BaseDAO.save(BaseDAO.java:88) ~[kitodo-data-management-3.3.jar:?]
	at org.kitodo.production.services.data.base.SearchDatabaseService.saveToDatabase(SearchDatabaseService.java:64) ~[classes/:3.3]
	at org.kitodo.production.services.data.base.SearchService.save(SearchService.java:278) ~[classes/:3.3]
	... 89 more
Caused by: org.hibernate.HibernateException: Don't change the reference to a collection with delete-orphan enabled : org.kitodo.data.database.beans.Workflow.templates
	at org.hibernate.engine.internal.Collections.prepareCollectionForUpdate(Collections.java:272) ~[hibernate-core-5.4.18.Final.jar:5.4.18.Final]
	at org.hibernate.engine.internal.Collections.processReachableCollection(Collections.java:231) ~[hibernate-core-5.4.18.Final.jar:5.4.18.Final]
	at org.hibernate.event.internal.FlushVisitor.processCollection(FlushVisitor.java:50) ~[hibernate-core-5.4.18.Final.jar:5.4.18.Final]
	at org.hibernate.event.internal.AbstractVisitor.processValue(AbstractVisitor.java:104) ~[hibernate-core-5.4.18.Final.jar:5.4.18.Final]
	at org.hibernate.event.internal.AbstractVisitor.processValue(AbstractVisitor.java:65) ~[hibernate-core-5.4.18.Final.jar:5.4.18.Final]
	at org.hibernate.event.internal.AbstractVisitor.processEntityPropertyValues(AbstractVisitor.java:59) ~[hibernate-core-5.4.18.Final.jar:5.4.18.Final]
	at org.hibernate.event.internal.DefaultFlushEntityEventListener.onFlushEntity(DefaultFlushEntityEventListener.java:182) ~[hibernate-core-5.4.18.Final.jar:5.4.18.Final]
	at org.hibernate.event.internal.AbstractFlushingEventListener.flushEntities(AbstractFlushingEventListener.java:232) ~[hibernate-core-5.4.18.Final.jar:5.4.18.Final]
	at org.hibernate.event.internal.AbstractFlushingEventListener.flushEverythingToExecutions(AbstractFlushingEventListener.java:92) ~[hibernate-core-5.4.18.Final.jar:5.4.18.Final]
	at org.hibernate.event.internal.DefaultFlushEventListener.onFlush(DefaultFlushEventListener.java:39) ~[hibernate-core-5.4.18.Final.jar:5.4.18.Final]
	at org.hibernate.event.service.internal.EventListenerGroupImpl.fireEventOnEachListener(EventListenerGroupImpl.java:102) ~[hibernate-core-5.4.18.Final.jar:5.4.18.Final]
	at org.hibernate.internal.SessionImpl.doFlush(SessionImpl.java:1360) ~[hibernate-core-5.4.18.Final.jar:5.4.18.Final]
	at org.hibernate.internal.SessionImpl.flush(SessionImpl.java:1347) ~[hibernate-core-5.4.18.Final.jar:5.4.18.Final]
	at org.kitodo.data.database.persistence.BaseDAO.storeObject(BaseDAO.java:351) ~[kitodo-data-management-3.3.jar:?]
	at org.kitodo.data.database.persistence.BaseDAO.save(BaseDAO.java:88) ~[kitodo-data-management-3.3.jar:?]
	at org.kitodo.production.services.data.base.SearchDatabaseService.saveToDatabase(SearchDatabaseService.java:64) ~[classes/:3.3]
	at org.kitodo.production.services.data.base.SearchService.save(SearchService.java:278) ~[classes/:3.3]
	... 89 more
```

--- a/Kitodo-DataManagement/src/main/java/org/kitodo/data/database/beans/BaseBean.java
+++ b/Kitodo-DataManagement/src/main/java/org/kitodo/data/database/beans/BaseBean.java
@@ -28,7 +28,7 @@ import org.kitodo.data.database.persistence.BaseDAO;
  * Base bean class.
  */
 @MappedSuperclass
-public abstract class BaseBean implements Cloneable, Serializable {
+public abstract class BaseBean implements Serializable {
 
     @Id
     @Column(name = "id")
@@ -48,12 +48,5 @@ public abstract class BaseBean implements Cloneable, Serializable {
         if (Objects.nonNull(this.id) && !Hibernate.isInitialized(list)) {
             baseDAO.initialize(this, list);
         }
-    }
-
-    @Override
-    public BaseBean clone() throws CloneNotSupportedException {
-        BaseBean clone = (BaseBean) super.clone();
-        clone.id = null;
-        return clone;
     }
 }

--- a/Kitodo-DataManagement/src/main/java/org/kitodo/data/database/beans/BaseIndexedBean.java
+++ b/Kitodo-DataManagement/src/main/java/org/kitodo/data/database/beans/BaseIndexedBean.java
@@ -22,7 +22,7 @@ import org.kitodo.data.database.enums.IndexAction;
  * Base bean class.
  */
 @MappedSuperclass
-public abstract class BaseIndexedBean extends BaseBean implements Cloneable {
+public abstract class BaseIndexedBean extends BaseBean {
 
     @Column(name = "indexAction")
     @Enumerated(EnumType.STRING)
@@ -46,12 +46,5 @@ public abstract class BaseIndexedBean extends BaseBean implements Cloneable {
      */
     public void setIndexAction(IndexAction indexAction) {
         this.indexAction = indexAction;
-    }
-
-    @Override
-    public BaseIndexedBean clone() throws CloneNotSupportedException {
-        BaseIndexedBean clone = (BaseIndexedBean) super.clone();
-        clone.indexAction = null;
-        return clone;
     }
 }

--- a/Kitodo-DataManagement/src/main/java/org/kitodo/data/database/beans/BaseTemplateBean.java
+++ b/Kitodo-DataManagement/src/main/java/org/kitodo/data/database/beans/BaseTemplateBean.java
@@ -20,7 +20,7 @@ import javax.persistence.MappedSuperclass;
  * This bean contains properties common for Template and Process.
  */
 @MappedSuperclass
-public abstract class BaseTemplateBean extends BaseIndexedBean implements Cloneable {
+public abstract class BaseTemplateBean extends BaseIndexedBean {
 
     @Column(name = "title")
     protected String title;
@@ -92,12 +92,5 @@ public abstract class BaseTemplateBean extends BaseIndexedBean implements Clonea
     @Override
     public String toString() {
         return title + " [" + id + ']';
-    }
-
-    @Override
-    public BaseTemplateBean clone() throws CloneNotSupportedException {
-        BaseTemplateBean clone = (BaseTemplateBean) super.clone();
-        clone.creationDate = (Date) creationDate.clone();
-        return clone;
     }
 }

--- a/Kitodo-DataManagement/src/main/java/org/kitodo/data/database/beans/Folder.java
+++ b/Kitodo-DataManagement/src/main/java/org/kitodo/data/database/beans/Folder.java
@@ -32,6 +32,7 @@ import javax.persistence.Transient;
 import org.kitodo.api.imagemanagement.ImageManagementInterface;
 import org.kitodo.config.ConfigMain;
 import org.kitodo.data.database.enums.LinkingMode;
+import org.kitodo.forms.FolderGenerator;
 
 /**
  * Stores configuration settings regarding a type of sub-folder in the process
@@ -60,7 +61,7 @@ import org.kitodo.data.database.enums.LinkingMode;
  */
 @Entity
 @Table(name = "folder")
-public class Folder extends BaseBean implements Cloneable {
+public class Folder extends BaseBean {
     /**
      * Default {@code fileGrp}s supported by the DFG viewer. The list is used to
      * populate a combo box in the edit dialog.
@@ -487,10 +488,5 @@ public class Folder extends BaseBean implements Cloneable {
     @Override
     public String toString() {
         return path + (path.isEmpty() || fileGroup.isEmpty() ? "" : ", ") + fileGroup;
-    }
-
-    @Override
-    public Folder clone() throws CloneNotSupportedException {
-        return (Folder) super.clone();
     }
 }

--- a/Kitodo-DataManagement/src/main/java/org/kitodo/data/database/beans/Project.java
+++ b/Kitodo-DataManagement/src/main/java/org/kitodo/data/database/beans/Project.java
@@ -30,7 +30,7 @@ import org.kitodo.data.database.persistence.ProjectDAO;
 
 @Entity
 @Table(name = "project")
-public class Project extends BaseIndexedBean implements Cloneable, Comparable<Project> {
+public class Project extends BaseIndexedBean implements Comparable<Project> {
 
     @Column(name = "title", nullable = false, unique = true)
     private String title;
@@ -492,52 +492,5 @@ public class Project extends BaseIndexedBean implements Cloneable, Comparable<Pr
     @Override
     public int hashCode() {
         return this.title == null ? 0 : this.title.hashCode();
-    }
-
-    @Override
-    public Project clone() throws CloneNotSupportedException {
-        // The primitive and immutable fields have already been correctly
-        // initialized by the wonderful magic of Object.clone().
-        Project clone = (Project) super.clone();
-
-        // Since date objects are mutable, they must be cloned. Otherwise
-        // changing the date on one project would also change the same date on
-        // the other project.
-        clone.startDate = (Date) startDate.clone();
-        clone.endDate = (Date) endDate.clone();
-
-        // The same goes for the list of assigned users and production
-        // templates. For Hibernate, these operations must be symmetric.
-        clone.users = new ArrayList<>(this.users);
-        for (User user : clone.users) {
-            user.getProjects().add(clone);
-        }
-
-        clone.templates = new ArrayList<>(this.templates);
-        for (Template template : clone.templates) {
-            template.getProjects().add(clone);
-        }
-
-        // The processes of the original project are not assigned to the clone
-        clone.processes = new ArrayList<>();
-
-        // The folder objects must also be cloned and the purposes must point to
-        // the cloned folders
-        clone.folders = new ArrayList<>();
-        for (Folder folder : this.folders) {
-            Folder clonedFolder = folder.clone();
-            clonedFolder.setProject(clone);
-            clone.folders.add(clonedFolder);
-            if (folder.equals(generatorSource)) {
-                clone.generatorSource = clonedFolder;
-            }
-            if (folder.equals(mediaView)) {
-                clone.mediaView = clonedFolder;
-            }
-            if (folder.equals(preview)) {
-                clone.preview = clonedFolder;
-            }
-        }
-        return clone;
     }
 }

--- a/Kitodo-DataManagement/src/main/java/org/kitodo/data/database/beans/Template.java
+++ b/Kitodo-DataManagement/src/main/java/org/kitodo/data/database/beans/Template.java
@@ -31,7 +31,7 @@ import org.kitodo.data.database.persistence.TemplateDAO;
 
 @Entity
 @Table(name = "template")
-public class Template extends BaseTemplateBean implements Cloneable {
+public class Template extends BaseTemplateBean {
 
     @Column(name = "active")
     private Boolean active = true;
@@ -260,23 +260,5 @@ public class Template extends BaseTemplateBean implements Cloneable {
     @Override
     public int hashCode() {
         return Objects.hash(client, docket, ruleset, workflow);
-    }
-
-    @Override
-    public Template clone() throws CloneNotSupportedException {
-        Template clone = (Template) super.clone();
-        clone.processes = new ArrayList<>();
-
-        // tasks don't need to be duplicated - will be created out of copied
-        // workflow
-        clone.tasks = new ArrayList<>();
-
-        clone.projects = new ArrayList<>(this.projects);
-        // make sure if copy should be assigned automatically to all projects
-        for (Project project : clone.projects) {
-            project.getTemplates().add(clone);
-        }
-
-        return clone;
     }
 }

--- a/Kitodo-DataManagement/src/main/java/org/kitodo/data/database/beans/Workflow.java
+++ b/Kitodo-DataManagement/src/main/java/org/kitodo/data/database/beans/Workflow.java
@@ -31,7 +31,7 @@ import org.kitodo.data.database.persistence.WorkflowDAO;
 
 @Entity
 @Table(name = "workflow")
-public class Workflow extends BaseIndexedBean implements Cloneable {
+public class Workflow extends BaseIndexedBean {
 
     @Column(name = "title")
     private String title;
@@ -181,10 +181,5 @@ public class Workflow extends BaseIndexedBean implements Cloneable {
     @Override
     public int hashCode() {
         return Objects.hash(title, status);
-    }
-
-    @Override
-    public Workflow clone() throws CloneNotSupportedException {
-        return (Workflow) super.clone();
     }
 }

--- a/Kitodo/src/main/java/org/kitodo/production/services/data/FolderService.java
+++ b/Kitodo/src/main/java/org/kitodo/production/services/data/FolderService.java
@@ -76,4 +76,29 @@ public class FolderService extends SearchDatabaseService<Folder, FolderDAO> {
         }
         return null;
     }
+
+    /**
+     * Creates a deep copy of a folder.
+     *
+     * @param source
+     *            folder to duplicate
+     *
+     * @return the duplicated folder
+     */
+    public Folder cloneFolder(Folder source) {
+        Folder clone = new Folder();
+        clone.setMimeType(source.getMimeType());
+        clone.setFileGroup(source.getFileGroup());
+        clone.setUrlStructure(source.getUrlStructure());
+        clone.setPath(source.getPath());
+        clone.setProject(source.getProject());
+        clone.setCopyFolder(source.isCopyFolder());
+        clone.setCreateFolder(source.isCreateFolder());
+        clone.setDerivative(source.getDerivative().orElse(null));
+        clone.setDpi(source.getDpi().orElse(null));
+        clone.setImageScale(source.getImageScale().orElse(null));
+        clone.setImageSize(source.getImageSize().orElse(null));
+        clone.setLinkingMode(source.getLinkingMode());
+        return clone;
+    }
 }

--- a/Kitodo/src/main/java/org/kitodo/production/services/data/ProjectService.java
+++ b/Kitodo/src/main/java/org/kitodo/production/services/data/ProjectService.java
@@ -12,7 +12,6 @@
 package org.kitodo.production.services.data;
 
 import java.io.IOException;
-import java.lang.reflect.UndeclaredThrowableException;
 import java.util.ArrayList;
 import java.util.Collections;
 import java.util.HashMap;
@@ -27,6 +26,7 @@ import org.elasticsearch.index.query.QueryBuilder;
 import org.elasticsearch.index.query.QueryBuilders;
 import org.kitodo.config.enums.KitodoConfigFile;
 import org.kitodo.data.database.beans.Client;
+import org.kitodo.data.database.beans.Folder;
 import org.kitodo.data.database.beans.Process;
 import org.kitodo.data.database.beans.Project;
 import org.kitodo.data.database.beans.Template;
@@ -256,13 +256,52 @@ public class ProjectService extends ClientSearchService<Project, ProjectDTO, Pro
      * @return the duplicated project
      */
     public Project duplicateProject(Project baseProject) {
-        try {
-            Project duplicatedProject = baseProject.clone();
-            duplicatedProject.setTitle("");
-            return duplicatedProject;
-        } catch (CloneNotSupportedException e) {
-            throw new UndeclaredThrowableException(e);
+        Project duplicatedProject = new Project();
+
+        // project title is intentionally not duplicated
+        duplicatedProject.setClient(baseProject.getClient());
+        duplicatedProject.setStartDate(baseProject.getStartDate());
+        duplicatedProject.setEndDate(baseProject.getEndDate());
+        duplicatedProject.setNumberOfPages(baseProject.getNumberOfPages());
+        duplicatedProject.setNumberOfVolumes(baseProject.getNumberOfVolumes());
+        duplicatedProject.setDmsImportRootPath(baseProject.getDmsImportRootPath());
+        duplicatedProject.setMetsRightsOwner(baseProject.getMetsRightsOwner());
+        duplicatedProject.setMetsRightsOwnerLogo(baseProject.getMetsRightsOwnerLogo());
+        duplicatedProject.setMetsRightsOwnerSite(baseProject.getMetsRightsOwnerSite());
+        duplicatedProject.setMetsRightsOwnerMail(baseProject.getMetsRightsOwnerMail());
+        duplicatedProject.setMetsDigiprovPresentation(baseProject.getMetsDigiprovPresentation());
+        duplicatedProject.setMetsDigiprovReference(baseProject.getMetsDigiprovReference());
+        duplicatedProject.setMetsPointerPath(baseProject.getMetsPointerPath());
+        duplicatedProject.setMetsPurl(baseProject.getMetsPurl());
+        duplicatedProject.setMetsContentIDs(baseProject.getMetsContentIDs());
+
+        FolderService folderService = ServiceManager.getFolderService();
+        List<Folder> duplicatedFolders = new ArrayList<>();
+        Folder generatorSource = null;
+        Folder mediaView = null;
+        Folder preview = null;
+
+        for (Folder folder : baseProject.getFolders()) {
+            Folder duplicatedFolder = folderService.cloneFolder(folder);
+            duplicatedFolder.setProject(duplicatedProject);
+            duplicatedFolders.add(duplicatedFolder);
+
+            if (folder.equals(baseProject.getGeneratorSource())) {
+                generatorSource = duplicatedFolder;
+            }
+            if (folder.equals(baseProject.getMediaView())) {
+                mediaView = duplicatedFolder;
+            }
+            if (folder.equals(baseProject.getPreview())) {
+                preview = duplicatedFolder;
+            }
         }
+        duplicatedProject.setFolders(duplicatedFolders);
+        duplicatedProject.setGeneratorSource(generatorSource);
+        duplicatedProject.setMediaView(mediaView);
+        duplicatedProject.setPreview(preview);
+
+        return duplicatedProject;
     }
 
     /**

--- a/Kitodo/src/main/java/org/kitodo/production/services/data/TemplateService.java
+++ b/Kitodo/src/main/java/org/kitodo/production/services/data/TemplateService.java
@@ -16,7 +16,6 @@ import java.io.FileInputStream;
 import java.io.FileNotFoundException;
 import java.io.IOException;
 import java.io.InputStream;
-import java.lang.reflect.UndeclaredThrowableException;
 import java.util.ArrayList;
 import java.util.Collections;
 import java.util.Date;
@@ -154,25 +153,28 @@ public class TemplateService extends ClientSearchService<Template, TemplateDTO, 
     }
 
     /**
-     * Duplicate a template.
+     * Duplicate the template with the given ID 'itemId'.
      *
-     * @param baseTemplate
-     *            template to duplicate
-     *
-     * @return the duplicated template
+     * @return the duplicated Template
      */
     public Template duplicateTemplate(Template baseTemplate) {
-        try {
-            Template duplicatedTemplate = baseTemplate.clone();
+        Template duplicatedTemplate = new Template();
 
-            // The title of the template is intentionally not copied
-            duplicatedTemplate.setTitle("");
-            duplicatedTemplate.setCreationDate(new Date());
+        // Template _title_ should explicitly _not_ be duplicated!
+        duplicatedTemplate.setCreationDate(new Date());
+        duplicatedTemplate.setClient(baseTemplate.getClient());
+        duplicatedTemplate.setDocket(baseTemplate.getDocket());
+        duplicatedTemplate.setRuleset(baseTemplate.getRuleset());
+        // tasks don't need to be duplicated - will be created out of copied workflow
+        duplicatedTemplate.setWorkflow(baseTemplate.getWorkflow());
 
-            return duplicatedTemplate;
-        } catch (CloneNotSupportedException e) {
-            throw new UndeclaredThrowableException(e);
+        // TODO: make sure if copy should be assigned automatically to all projects
+        for (Project project : baseTemplate.getProjects()) {
+            duplicatedTemplate.getProjects().add(project);
+            project.getTemplates().add(duplicatedTemplate);
         }
+
+        return duplicatedTemplate;
     }
 
     @Override
@@ -414,7 +416,7 @@ public class TemplateService extends ClientSearchService<Template, TemplateDTO, 
         BoolQueryBuilder query = new BoolQueryBuilder();
 
         if (Objects.nonNull(filters) && !filters.isEmpty()) {
-            Map<String, String> filterMap = filters;
+            Map<String, String> filterMap = (Map<String, String>) filters;
             query.must(readFilters(filterMap));
         }
         query.must(getQueryForSelectedClient());

--- a/Kitodo/src/main/java/org/kitodo/production/services/data/WorkflowService.java
+++ b/Kitodo/src/main/java/org/kitodo/production/services/data/WorkflowService.java
@@ -11,7 +11,6 @@
 
 package org.kitodo.production.services.data;
 
-import java.lang.reflect.UndeclaredThrowableException;
 import java.util.Collections;
 import java.util.List;
 import java.util.Map;
@@ -115,23 +114,20 @@ public class WorkflowService extends ClientSearchService<Workflow, WorkflowDTO, 
     }
 
     /**
-     * Duplicates the given workflow.
+     * Duplicate the given workflow.
      *
      * @param baseWorkflow
-     *            workflow to copy
-     * @return the duplicated workflow
+     *            to copy
+     * @return the duplicated Workflow
      */
     public Workflow duplicateWorkflow(Workflow baseWorkflow) {
-        try {
-            Workflow duplicatedWorkflow = baseWorkflow.clone();
+        Workflow duplicatedWorkflow = new Workflow();
 
-            duplicatedWorkflow.setTitle(baseWorkflow.getTitle() + "_" + Helper.generateRandomString(3));
-            duplicatedWorkflow.setStatus(WorkflowStatus.DRAFT);
+        duplicatedWorkflow.setTitle(baseWorkflow.getTitle() + "_" + Helper.generateRandomString(3));
+        duplicatedWorkflow.setStatus(WorkflowStatus.DRAFT);
+        duplicatedWorkflow.setClient(baseWorkflow.getClient());
 
-            return duplicatedWorkflow;
-        } catch (CloneNotSupportedException e) {
-            throw new UndeclaredThrowableException(e);
-        }
+        return duplicatedWorkflow;
     }
 
     /**

